File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string t, string p) {
    int ret = 0;
    int tLen = t.size();
    int pLen = p.size();
    
    for (int i=0; i<=tLen-pLen; i++) {
        if (t.substr(i, pLen) <= p) ret++;
    }
    return ret;
}

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.py
"""
- p의 길이 구하기
- p의 길이 만큼 t 부분 문자열 구하기
- 부분 문자열 대수 비교 후 카운트
"""

def solution(t, p):
    tLen, pLen = len(t), len(p)
    subList = [t[i:i+pLen] for i in range(tLen - pLen + 1)]
    
    ret = 0
    for sub in subList:
        if sub <= p:
            ret += 1

    return ret